projects/m5/002-polygon-area-calculator/js/shape_calculator.js

```diff
@@ -0,0 +1,112 @@
+class Rectangle {
+  #name = "Rectangle";
+
+  constructor(width, height) {
+    if (width < 0 || height < 0) {
+      console.log(
+        "Error!: The input must be a positive number\nthe height and width value has been set to 0"
+      );
+      this._width = 0;
+      this._height = 0;
+    } else {
+      this._width = width;
+      this._height = height;
+    }
+  }
+
+  getWidth() {
+    return this._width;
+  }
+
+  setWidth(value) {
+    if (value < 0) {
+      console.log(
+        `Error!: The input must be a positive number\nthe width has been set to the previous value (${this._width})`
+      );
+      value = this._width;
+    }
+    this._width = value;
```

Comment:
Other possible solution:
`value > 0  ?  this._width = value  :  this._width = 0`